de.dashup.application/src/main/java/de/dashup/application/controllers/MarketplaceController.java

```diff
@@ -19,7 +22,11 @@
     @RequestMapping("/")
     public String marketplace(@CookieValue(name = "token", required = false) String token, Model model, HttpServletRequest request) throws SQLException {
         return ControllerHelper.defaultMapping(token, request, model, "marketplace", user -> {
-            // tbd
+            model.addAttribute("carouselWidget",DashupService.getInstance().getPanelById(1));
+            int[] featuredWidgets = {1,2,3,4};
```

Comment:
We should change this to four random numbers from all widgets available lateron

Reply:
Yes lateron, but not now.... we agreed to hardcoded widgets in this section so I think u should not mention that here...

Reply:
I implemented the logic for you using the RAND() function of maria db in combination with LIMIT. Now you can enhance the amout of widgets in the featured section by changing one number.